### PR TITLE
Few fixies and feature proposals

### PR DESCRIPTION
**Pull request by xblackman**
Please see more in commit info

### CODE REVIEW

**Comment by edips** [comment]
Looks beautiful. Are you using ArcGIS Runtime library or Proj4 for coordinate system?

**Comment by xblackman** [comment]
Screenshots are taken from an application in early development. Map is shown just by Qt Location module (osm plugin). Proj4 will be used later for coordinates conversion.

File: src/demo/demo.pro [hunk 1]
```diff
@@ -0,0 +1,20 @@
+TEMPLATE = app
```
Comment (plfiorini) [inline src/demo/demo.pro]: This file is not needed the demo is built by qbs

File: src/demo/demo.qrc [hunk 2]
```diff
@@ -0,0 +1,44 @@
+<RCC>
```
Comment (plfiorini) [inline src/demo/demo.qrc]: This file is not needed the demo is built by qbs and the resources are generated by it automatically

File: src/imports/controls/DateTimePickerDialog.qml [hunk 3]
```diff
@@ -118,7 +118,7 @@ Dialog {
                 anchors.verticalCenter: parent.verticalCenter
                 text: qsTr("Now")
                 flat: true
-                onClicked: dateTimePickerDialog.selectedDate = new Date()
+                onClicked: dateTimePickerDialog.selectedDateTime = new Date()
             }
         }
         \endcode
```
Comment (plfiorini) [inline src/imports/controls/DateTimePickerDialog.qml]: both fixes are already in `develop`

File: src/imports/controls/NavigationDrawer.qml [hunk 4]
```diff
@@ -162,6 +162,7 @@ Drawer {
 
                 delegate: ListItem {
                     icon.name: modelData.icon.name
+                    icon.source: modelData.icon.source
```
Comment (plfiorini) [inline src/imports/controls/NavigationDrawer.qml]: This should be in its own commit

File: src/imports/controls/Page.qml [hunk 5]
```diff
@@ -154,7 +154,13 @@ Page {
     FluidControls.AppBar {
```
Comment (plfiorini) [inline src/imports/controls/Page.qml]: This should be in its own commit

File: src/imports/controls/ListItem.qml [hunk 6]
```diff
@@ -92,6 +92,13 @@ ItemDelegate {
     */
```
Comment (plfiorini) [inline src/imports/controls/ListItem.qml]: Why do you need to add the `keepMissingIconSpace` property instead of doing:

```
ListItem {
    leftItem: Item {
        width: 40
        height: width
    }
}
```

File: src/imports/controls/InfoBar.qml [hunk 7]
```diff
@@ -141,6 +141,10 @@ Rectangle {
         }
```
Comment (plfiorini) [inline src/imports/controls/InfoBar.qml]: This should be in its own commit

File: src/imports/controls/ListItem.qml (hunk 6 above)
Comment (xblackman) [reply src/imports/controls/ListItem.qml]: Do you mean placing the "leftItem" workaround in my code ? Currently I use "keepMissingIconSpace" in my code as follows:

 delegate: FluidControls.ListItem {
            id: listItem
            text: model.text
            highlighted: ListView.isCurrentItem
            keepMissingIconSpace: true  // NEW, proposed property
            icon.source: ListView.isCurrentItem ? FluidControls.Utils.iconUrl("navigation/check") : ""

...I do not also need to know the item width.

File: src/demo/demo.qrc (hunk 2 above)
Comment (xblackman) [reply src/demo/demo.qrc]: If I am not using qbs ? ... a part of the project can be built using qmake (the library itself), but I need qbs for building the rest (the demo). I would suggest to add full support for qmake across the whole project or support JUST qbs.

File: src/imports/controls/DateTimePickerDialog.qml (hunk 3 above)
Comment (xblackman) [reply src/imports/controls/DateTimePickerDialog.qml]: I made a fork from develop.

File: src/demo/demo.qrc (hunk 2 above)
Comment (plfiorini) [reply src/demo/demo.qrc]: the qmake files are here only to include the library in qmake projects and you don't need the demo in that case.

File: src/imports/controls/NavigationDrawer.qml (hunk 4 above)
Comment (plfiorini) [reply src/imports/controls/NavigationDrawer.qml]: you can make a single pull request with multiple commits. each change should be separate and have a detailed commit message

File: src/demo/demo.qrc (hunk 2 above)
Comment (xblackman) [reply src/demo/demo.qrc]: Ok, understood.

File: src/imports/controls/NavigationDrawer.qml (hunk 4 above)
Comment (xblackman) [reply src/imports/controls/NavigationDrawer.qml]: I thought that I did exactly what you just said. I made 7 commits, each with own description. The description was not sufficient ? Next time, I will be smarter, just instruct me how to do that next time according to your wishes.

Comment (plfiorini) [reply src/imports/controls/NavigationDrawer.qml]: nvm my mistake the PR email from github misled me, i can cherry pick the relevan commits then.

File: src/imports/controls/Page.qml [hunk 8]
```diff
@@ -154,7 +154,13 @@ Page {
...
         id: appBar
 
+        padding: 0
         Material.elevation: 0
+        anchors {
+          left: parent.left
+          top: parent.top
+          right: parent.right
+        }
```
Comment (plfiorini) [inline src/imports/controls/Page.qml]: the app bar is not anchored on purpose, it is used to replace the toolbar when a new page is pushed

File: src/imports/controls/ListItem.qml (hunk 6 above)
Comment (plfiorini) [reply src/imports/controls/ListItem.qml]: Have you tried a `CheckDelegate` for `leftItem`?

Comment (xblackman) [reply src/imports/controls/ListItem.qml]: No, I will. ...I need to reach the behaviour shown in the pictures above without too much workaround.

File: src/imports/controls/Page.qml (hunk 8 above)
Comment (xblackman) [reply src/imports/controls/Page.qml]: Can I reach desired state as shown in the screnshots without my modification ?

**Comment by xblackman** [comment]
As I understood InfoBar is now SnackBar.

File: src/imports/controls/Page.qml (hunk 8 above)
Comment (plfiorini) [reply src/imports/controls/Page.qml]: Hard to tell since I don't know how does it look without the anchors and I didn't see the code.

**Comment by plfiorini** [comment]
@xblackman yes but your change is already cherry-picked

File: src/imports/controls/NavigationDrawer.qml (hunk 4 above)
Comment (plfiorini) [reply src/imports/controls/NavigationDrawer.qml]: cherry-picked in `develop`

File: src/imports/controls/Page.qml (hunk 8 above)
Comment (xblackman) [reply src/imports/controls/Page.qml]: I use a Popup opened from the Action assigned to the Settings icon 
The Popup is defined as follows>

    Popup {
        id: settingsPopup
        x: parent.width - width
        y: 0
        width: 400
        height: 350
        padding: 0
        
        FluidControls.PageStack {
            id: settingsStack
            anchors.fill: parent
            initialItem: QuickSettings {}
            clip: true
        }        
    }

Without suggested modification the setting page shows without a title as shown in the picture.
image
image
